language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/KeySpecifierNodeContext.java

```diff
@@ -89,23 +92,49 @@ public KeySpecifierNodeContext() {
                 return completionItems;
             }
             rowTypeSymbol = CommonUtil.getRawType(typeSymbol.get());
+            // If the context type is again a table, need to derive row type.
+            // This is valid for cases where table type is defined separately as a user defined type.
+            // ex:
+            //      type PersonTable table<Person> key(name);
+            //      PersonTable tbl = table key(<cursor>);
+            if (rowTypeSymbol.typeKind() == TypeDescKind.TABLE) {
+                TableTypeSymbol tableTypeSymbol = (TableTypeSymbol) rowTypeSymbol;
+                rowTypeSymbol = CommonUtil.getRawType(tableTypeSymbol.rowTypeParameter());
+            }
```

Comment:
Can't we handle this within the `ContextTypeResolver`? Also, if we look at the outer-most `if` clause, we check the type of the node's parent and it is better if we can get the context type from the resolver itself.